### PR TITLE
tests: GH421 vkUpdateDescriptorSets tests

### PR DESCRIPTION
**Pull request by MarkMLunarG**
A) Adding the WriteDescriptorSetIntegrityCheck test, which verifies
some requirements of chapter 13.2.3 of the Vulkan Spec
 1) A uniform buffer update must have a valid buffer index.
 2) When using an array of descriptors in a single WriteDescriptor,
      the descriptor types and stageflags must all be the same.
 3) Immutable Sampler state must match across descriptors
The above tests also exercise cases where descriptors are updated with
consecutive bindings both correctly and incorrectly

B) adding a MissingClearAttachment test that points to a wrong
colorAttachment index in a VkClearAttachment structure passed to
vkCmdClearAttachments

C) update vk_validation_layer_details.md to reflect above changes.

Change-Id: I3f4f2233371b68caf23b5a4111d63a94765121c3

### CODE REVIEW

File: layers/vk_validation_layer_details.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ The Draw State portion of the core validation layer tracks state leading into Dr
 | Valid Secondary CommandBuffer | Validates that no primary command buffers are sent to vkCmdExecuteCommands() are | INVALID_SECONDARY_COMMAND_BUFFER | vkCmdExecuteCommands | ExecuteCommandsPrimaryCB | NA |
 | Invalid Descriptor Set | Invalid Descriptor Set used. Either never created or already destroyed. | INVALID_DESCRIPTOR_SET | vkQueueSubmit | TODO | Create Test |
 | Descriptor Type | Verify Descriptor type in bound descriptor set layout matches descriptor type specified in update. This also includes mismatches in the TYPES of copied descriptors. | DESCRIPTOR_TYPE_MISMATCH | vkUpdateDescriptorSets | DSTypeMismatch CopyDescriptorUpdateErrors | NA |
-| Descriptor StageFlags | Verify all descriptors within a single write update have the same stageFlags | DESCRIPTOR_STAGEFLAGS_MISMATCH | vkUpdateDescriptorSets | TODO | Test this case |
+| Descriptor StageFlags | Verify all descriptors within a single write update have the same stageFlags | DESCRIPTOR_STAGEFLAGS_MISMATCH | vkUpdateDescriptorSets | WriteDescriptorSetIntegrityCheck | Test this case |
```
Comment (tobine) [inline layers/vk_validation_layer_details.md]: Can we now remove the "Test this case" note?

File: layers/vk_validation_layer_details.md [hunk 2]
```diff
@@ -79,11 +79,11 @@ The Draw State portion of the core validation layer tracks state leading into Dr
...
 | Valid sampler descriptor Updates | An invalid sampler is used when updating SAMPLER descriptor. | SAMPLER_DESCRIPTOR_ERROR | vkUpdateDescriptorSets | SampleDescriptorUpdateError | Currently only making sure sampler handle is known, can add further validation for sampler parameters |
-| Immutable sampler update consistency | Within a single write update, all sampler updates must use either immutable samplers or non-immutable samplers, but not a combination of both. | INCONSISTENT_IMMUTABLE_SAMPLER_UPDATE | vkUpdateDescriptorSets | TODO | Write a test for this case |
+| Immutable sampler update consistency | Within a single write update, all sampler updates must use either immutable samplers or non-immutable samplers, but not a combination of both. | INCONSISTENT_IMMUTABLE_SAMPLER_UPDATE | vkUpdateDescriptorSets | WriteDescriptorSetIntegrityCheck | Write a test for this case |
```
Comment (tobine) [inline layers/vk_validation_layer_details.md]: Can we now remove the "Write a test this case" note?

File: layers/vk_validation_layer_details.md [hunk 3]
```diff
@@ -79,11 +79,11 @@ The Draw State portion of the core validation layer tracks state leading into Dr
...
 | Valid bufferView descriptor Updates | An invalid bufferView is used when updating *_TEXEL_BUFFER descriptor. | BUFFERVIEW_DESCRIPTOR_ERROR | vkUpdateDescriptorSets | InvalidBufferViewObject | Currently only making sure bufferView handle is known, can add further validation for bufferView parameters |
-| Valid bufferInfo descriptor Updates | An invalid bufferInfo is used when updating *_UNIFORM_BUFFER* or *_STORAGE_BUFFER* descriptor. | BUFFERINFO_DESCRIPTOR_ERROR | vkUpdateDescriptorSets | TODO | Implement validation test |
-| Attachment References in Subpass | Attachment reference must be present in active subpass | MISSING_ATTACHMENT_REFERENCE | vkCmdClearAttachments | TODO | Currently only making sure bufferInfo has buffer whose handle is known, can add further validation for bufferInfo parameters |
+| Valid bufferInfo descriptor Updates | An invalid bufferInfo is used when updating *_UNIFORM_BUFFER* or *_STORAGE_BUFFER* descriptor. | BUFFERINFO_DESCRIPTOR_ERROR | vkUpdateDescriptorSets | WriteDescriptorSetIntegrityCheck | Implement validation test |
```
Comment (tobine) [inline layers/vk_validation_layer_details.md]: Can we now remove the "Implement validation test" note?

**Comment by tobine** [comment]
Test code looks good. Just a couple of doc updates and should be good-to-push.
:+1:

**Comment by MarkMLunarG** [comment]
Completed with SHA-1: 9d2e7a3

**Pull request merged**